aips/move-spec-verifier.md

```diff
@@ -0,0 +1,114 @@
+
+1. Binary operator replacement - replaces binary operators with other binary operators. For example, the `+` operator can be replaced with the `-` operator. Operators are grouped into the following categories (operators are replaced within the same category. For example, the `+` operator can be replaced with the `-` operator but not the `<<` operator):
+   - arithmetic operators: `+`, `-`, `*`, `/`, `%`
+   - bitwise operators: `&`, `|`, `^`
+   - shifts: `<<`, `>>`
+   - comparison operators: `==`, `!=`, `<`, `>`, `<=`, `>=`
+   - logical operators: `&&`, `||`
+2. Unary operator replacement - this mutation operator replaces unary operators with other unary operators. For example, the `!` operator can be replaced with the space.
+3. Type replacement - replaces types with other types. For example, the `u8` type can be replaced with the `u64` type.
+4. Literal replacement - replaces literals with other literals. For example, the `0` literal can be replaced with the `1` literal or other random literal, `true` to `false`, etc. It's possible to choose the type of the literal to be replaced. For example, it's possible to replace only boolean literals.
+5. Address replacement - replaces addresses with other addresses. For example, the `0x1`/`@std` address can be replaced with the `0x000A` address or another random address.
+6. Return value replacement - replaces return values with other return values. For example, the concrete expressions can be replaced with concrete literals or other random literals.
+7. Break/continue replacement or deletion - replaces or deletes break/continue statements with other break/continue statements.
+
+The `spec-verify` command is responsible for running the Move mutator tool and then checking if the generated mutants are valid according to the Move specifications. It will do the following:
+1. Take command line arguments both for the Move Prover tool and for the Move mutator tool. It can also read the configuration from the JSON configuration file.
+2. Run the Move mutator tool (`mutate`) to generate mutants with the previously specified parameters.
+3. Run the Move Prover tool, passing the generated mutants individually.
+4. Collect the results and generate a report.
+
+The report contains information about the generated mutants as well as the killed ones for each tested source file. All the intermediate results are saved in the configurable (default: `mutants_output`) directory.
+
+## Reference Implementation
+
+TBD
```

Comment:
You can provide a link here to the tool under development (and mention that it is a work in progress).

Reply:
Added link.